vfs/vfscache/cache.go

```diff
@@ -75,17 +81,21 @@ type AddVirtualFn func(remote string, size int64, isDir bool) error
 // context passed in.
 func New(ctx context.Context, fremote fs.Fs, opt *vfscommon.Options, avFn AddVirtualFn) (*Cache, error) {
 	fName := fremote.Name()
-	fRoot := filepath.FromSlash(fremote.Root())
+	fRoot := fremote.Root()
 	if runtime.GOOS == "windows" {
-		if strings.HasPrefix(fRoot, `\\?`) {
-			fRoot = fRoot[3:]
-		}
-		fRoot = strings.Replace(fRoot, ":", "", -1)
-		// Replace leading ':' if remote was created on the fly as ":backend:/path" as it is illegal in Windows
+		// Make remote name valid as dir name below cache dir.
```

Comment:
Why don't we just use the encoder.OS encoding here?

Reply:
Could definitely do that. I *was* thinking about it, but decided not to touch that part..
All "on the fly" remotes would then be in cache folders starting with the `fullwidth colon`, which looks a bit "alien", but does probably not matter since it is just a cache folder for "internal use".